go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,328 @@
+	// example output from splitStdout
+	// splitStdout = {
+	// "158b96bc-b60b-4fff-b838-f8a5c75af557",
+	// "ip4.src==10.244.1.2",
+	// "",
+	// "0870c6b4-56e4-415f-967d-dde300266931",
+	// "ip4.src==10.244.2.2",
+	// "",
+	// "f10ce608-6286-4f93-9a1f-8b1d92494e17",
+	// "ip4.dst == 0.0.0.0/0 && ip4.src == $a5154718082306775057",
+	// "",
+	// "a785e260-086e-4ac4-b6bf-e2d3cceddce7",
+	// "ip4.src==10.244.0.2",
+	// "",
+	// "53cab5de-5e5b-405c-9915-b1efec69d433",
+	// "ip4.dst == 1.2.3.0/24 && ip4.src == $a5154718082306775057",
+	// }
+	// each pair of values is a single represents a single ACL entry
+	// 158b96bc-b60b-4fff-b838-f8a5c75af557 is the uuid and
+	// ip4.src==10.244.1.2 is the match statment
+	//
+	// in order to process them correctly loop by twos and if the current line is
+	// an empty string skip it, the current index is the UUID of the acl and
+	// index++ will tell us if the rule uses  ipv4 or ipv6 addresses
+	for index := 0; index < len(splitStdout); index = index + 2 {
+		if splitStdout[index] == "" {
+			// between every entry there is a newline, skip the newline
+			index++
+		}
+		uuid := splitStdout[index]
+		var addressSetHash string
+		if strings.Contains(splitStdout[index+1], "ip4") {
+			addressSetHash = nsInfo.addressSet.GetIPv4HashName()
+		} else {
+			addressSetHash = nsInfo.addressSet.GetIPv6HashName()
+		}
+
+		for _, node := range existingNodes.Items {
+			_, stderr, err := util.RunOVNNbctl("remove", "logical_switch",
+				joinSwitch(node.Name), "acls", uuid)
+			if err != nil {
+				errList = append(errList, fmt.Errorf("failed to delete the rules for "+
+					"address_set=%s, stderr: %q (%v)", addressSetHash, stderr, err))
+			}
+
+		}
+	}
+	return errList
+}
+
+func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewallapi.EgressFirewall) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		return oc.kube.UpdateEgressFirewall(egressfirewall)
+	})
+}
+
+func (ef *egressFirewall) addACLToJoinSwitch(joinSwitch, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string) error {
+	for _, rule := range ef.egressRules {
+		var match string
+		var action string
+		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
+			action = "allow"
+		} else {
+			action = "drop"
+		}
+
+		ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+		if err != nil {
+			return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+		}
+		if !utilnet.IsIPv6(ipAddress) {
+			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+		} else {
+			match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+		}
+
+		if len(rule.ports) > 0 {
+
+			match = fmt.Sprintf("%s && ( %s )\"", match[:len(match)-1], egressGetL4Match(rule.ports))
+		}
+		uuid, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
```

Comment:
maybe I already asked this, but can we use the golang bindings here? I know this is a time crunch, but in the future we should all be using the bindings and not doing ovn-nbctl commands anymore.

Reply:
we will take this on in a follow up